app/services/sla_breach_service.rb

```diff
@@ -11,11 +11,27 @@ def breach_threshold_date
   # attention_needed_since is present and is before breach threshold
   # returns data in the format { vita_partner_id: breach_count }
   def attention_needed_breaches
-    Client.sla_tracked.group(:vita_partner_id).where(Client.arel_table[:attention_needed_since].lteq(breach_threshold_date)).count(:id)
+    Client.sla_tracked
+      .where(Client.arel_table[:attention_needed_since].lteq(breach_threshold_date))
+      .group(:vita_partner_id)
+      .count(:id)
   end
 
   # clients who messaged us and have not been responded to _with a message, call or email_ within the breach threshold
   def outgoing_communication_breaches
-    Client.sla_tracked.group(:vita_partner_id).where(Client.arel_table[:first_unanswered_incoming_interaction_at].lteq(breach_threshold_date)).count(:id)
+    Client.sla_tracked
+      .where(Client.arel_table[:first_unanswered_incoming_interaction_at].lteq(breach_threshold_date))
+      .group(:vita_partner_id)
+      .count(:id)
+  end
+
+  # clients who've messaged us _and we've not interacted with their profile at all_ within the breach threshold.
+  def outgoing_interaction_breaches
+    clients = Client.arel_table
+    Client.sla_tracked
+      .where(clients[:first_unanswered_incoming_interaction_at].lteq(breach_threshold_date))
+      .where(clients[:last_interaction_at].lt(clients[:first_unanswered_incoming_interaction_at]).or(clients[:last_interaction_at].eq(nil)))
+      .group(:vita_partner_id)
+      .count(:id)
```

Comment:
I'm having a little trouble understanding this query, so I'll try to confirm if I understand it correctly.
It will return counts of clients, for each vita partner, where we are in breach of SLA (client has taken some key actions and we haven't responded in over 3 business days) AND there are no other key hub user interaction after that time.

Is `last_interaction_at` always set to nil when a client takes a key action?

Reply:
In which situations would a client show up in `outgoing_communication_breaches` but not in `outgoing_interaction_breaches`?
Are there any situations when a client would show up in `outgoing_interaction_breaches` but not `outgoing_communication_breaches`?

Reply:
Is `first_unanswered_incoming_interaction_at` set to nil when a volunteer takes a key action, the same as `attention_needed_since`?

Reply:
https://github.com/codeforamerica/vita-min/blob/main/app/models/concerns/interaction_tracking.rb